langchain/sql_database.ts

```diff
@@ -0,0 +1,99 @@
+import { DataSource } from "typeorm";
+import {
+    generateTableInfoFromTables,
+    getTableAndColumnsName,
+    verifyIgnoreTablesExistInDatabase,
+    verifyIncludeTablesExistInDatabase,
+    verifyListTablesExistInDatabase
+} from './util/sql_utils';
+
+export interface SQLDatabaseParams {
+    appDataSource: DataSource;
+    includesTables?: Array<string>;
+    ignoreTables?: Array<string>;
+    sampleRowsInTableInfo?: number;
+}
+
+export interface SqlTable {
+    tableName: string;
+    columns: SqlColumn[];
+}
+
+export interface SqlColumn {
+    columnName: string;
+    dataType?: string;
+    isNullable?: boolean;
+}
+
+export class SQLDatabase {
+    appDataSource: DataSource;
+
+    allTables: Array<SqlTable> = [];
```

Comment:
I chose to store directly table and columns names as we are retrieving it at the beginning